git-sync/main.go

```diff
@@ -213,3 +227,30 @@ func setupGitAuth(username, password, gitURL string) error {
 
 	return nil
 }
+
+func setupGitSSH() error {
+	log.Println("setting up git SSH credentials")
+
+	if _, err := os.Stat("/etc/git-secret/id-rsa"); err != nil {
+		return fmt.Errorf("error: could not find SSH secret: %v", err)
+	}
+
+	// Secret permissions must be restricted to use it as SSH key
+	if err := os.Chmod("/etc/git-secret/id-rsa", 0400); err != nil {
```

Comment:
Is it worth checking if the file is already 0400? I realize kube probably does whatever it does which is never 0400.